packages/arb-ts/src/lib/message/L1Transaction.ts

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2021, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-env node */
+'use strict'
+
+import { TransactionReceipt } from '@ethersproject/providers'
+import { Log } from '@ethersproject/abstract-provider'
+import { ContractTransaction } from '@ethersproject/contracts'
+import { BigNumber } from '@ethersproject/bignumber'
+import {
+  L1ToL2Message,
+  L1ToL2MessageReaderOrWriter,
+  L1ToL2MessageReader,
+  L1ToL2MessageWriter,
+} from './L1ToL2Message'
+
+import { L1ERC20Gateway__factory, Inbox__factory } from '../abi'
+import { DepositInitiatedEvent } from '../abi/L1ERC20Gateway'
+import {
+  SignerProviderUtils,
+  SignerOrProvider,
+} from '../dataEntities/signerOrProvider'
+import { ArbTsError } from '../dataEntities/errors'
+
+export interface L1ContractTransaction extends ContractTransaction {
+  wait(confirmations?: number): Promise<L1TransactionReceipt>
+}
+
+export class L1TransactionReceipt implements TransactionReceipt {
+  public readonly to: string
+  public readonly from: string
+  public readonly contractAddress: string
+  public readonly transactionIndex: number
+  public readonly root?: string
+  public readonly gasUsed: BigNumber
+  public readonly logsBloom: string
+  public readonly blockHash: string
+  public readonly transactionHash: string
+  public readonly logs: Array<Log>
+  public readonly blockNumber: number
+  public readonly confirmations: number
+  public readonly cumulativeGasUsed: BigNumber
+  public readonly effectiveGasPrice: BigNumber
+  public readonly byzantium: boolean
+  public readonly type: number
+  public readonly status?: number
+
+  constructor(tx: TransactionReceipt) {
+    this.to = tx.to
+    this.from = tx.from
+    this.contractAddress = tx.contractAddress
+    this.transactionIndex = tx.transactionIndex
+    this.root = tx.root
+    this.gasUsed = tx.gasUsed
+    this.logsBloom = tx.logsBloom
+    this.blockHash = tx.blockHash
+    this.transactionHash = tx.transactionHash
+    this.logs = tx.logs
+    this.blockNumber = tx.blockNumber
+    this.confirmations = tx.confirmations
+    this.cumulativeGasUsed = tx.cumulativeGasUsed
+    this.effectiveGasPrice = tx.effectiveGasPrice
+    this.byzantium = tx.byzantium
+    this.type = tx.type
+    this.status = tx.status
+  }
+
+  /**
+   * Get the numbers of any messages created by this transaction
+   * @returns
+   */
+  public getMessageNumbers(): BigNumber[] {
```

Comment:
hm I'd argue its cleaner if we change this to `getInboxMessages` and return the event.
Its pretty easy for the consumer to then do: `receipt.getInboxMessages().messageNum`

Reply:
would be consistent with
`public getDepositEvents(): DepositInitiatedEvent['args'][]`

Reply:
Updated